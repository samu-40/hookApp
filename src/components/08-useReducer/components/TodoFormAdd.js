import React from 'react';
import PropTypes from 'prop-types';
import { useForm } from '../../../hooks/useForm';

const TodoFormAdd = ({handleAdd}) => {

    const [{description}, handleInputChanges, reset] = useForm({description:''});

    const handleSubmit = (e) => {
        e.preventDefault();

        if (description.trim().length <= 1) {return};

        const newTodo = {
            id: new Date().getTime(),
            desc: description,
            done: false,
        };

        handleAdd(newTodo);

        reset();

    };


    return (
        <>
            <h6>Agregar TODO</h6>
            <hr />

            <form
                className='d-grid gap-2'
                onSubmit={handleSubmit}
            >
                <input
                    type='text'
                    name='description'
                    placeholder='TODO....'
                    autoComplete='off'
                    className='form-control'
                    value={description}
                    onChange={handleInputChanges}
                />

                <button
                    type='submit'
                    className='btn btn-outline-success mt-1 '
                >
                    Add
                        </button>
            </form>
        </>
    );
};

TodoFormAdd.propTypes = {
    handleAdd: PropTypes.func.isRequired,
}

export default TodoFormAdd
