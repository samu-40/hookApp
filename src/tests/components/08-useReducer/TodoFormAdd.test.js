import {shallow} from 'enzyme';
import TodoFormAdd from "../../../components/08-useReducer/components/TodoFormAdd";

describe('Pruebas en TodoFormAdd.js', () => {

    const handleAdd = jest.fn();

    const wrapper = shallow(<TodoFormAdd handleAdd={handleAdd}/>);

    test('Mostrar el componente correctamente', () => {
        expect(wrapper).toMatchSnapshot();
    });

    test('No se debe llamar a handleAdd', () => {
        const formSubmit = wrapper.find('form').prop('onSubmit');

        formSubmit({preventDefault(){}});

        expect(handleAdd).toHaveBeenCalledTimes(0);
    });

    test('Se debe llamar a handleAdd con un objeto como parametro y resetear el input', () => {
        const value = 'Aprender GitHub';

        wrapper.find('input').simulate('change', {
            target:{
                value, name: 'description'
            }
        });

        const formSubmit = wrapper.find('form').prop('onSubmit'); 

        formSubmit({preventDefault(){}});

        expect(handleAdd).toHaveBeenCalledTimes(1);
        expect(handleAdd).toHaveBeenCalledWith({
            id: expect.any(Number),
            desc: value,
            done: false,
        });
        expect(wrapper.find('input').prop('value')).toBe(''); 
    });
    
});