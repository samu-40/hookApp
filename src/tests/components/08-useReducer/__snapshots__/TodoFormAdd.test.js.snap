// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas en TodoFormAdd.js Mostrar el componente correctamente 1`] = `
<Fragment>
  <h6>
    Agregar TODO
  </h6>
  <hr />
  <form
    className="d-grid gap-2"
    onSubmit={[Function]}
  >
    <input
      autoComplete="off"
      className="form-control"
      name="description"
      onChange={[Function]}
      placeholder="TODO...."
      type="text"
      value=""
    />
    <button
      className="btn btn-outline-success mt-1 "
      type="submit"
    >
      Add
    </button>
  </form>
</Fragment>
`;
